src/main/java/org/cryptomator/ui/updatereminder/UpdateReminderModule.java

```diff
@@ -0,0 +1,59 @@
+package org.cryptomator.ui.updatereminder;
+
+import dagger.Binds;
+import dagger.Module;
+import dagger.Provides;
+import dagger.multibindings.IntoMap;
+import org.cryptomator.ui.common.DefaultSceneFactory;
+import org.cryptomator.ui.common.FxController;
+import org.cryptomator.ui.common.FxControllerKey;
+import org.cryptomator.ui.common.FxmlFile;
+import org.cryptomator.ui.common.FxmlLoaderFactory;
+import org.cryptomator.ui.common.FxmlScene;
+import org.cryptomator.ui.common.StageFactory;
+
+import javax.inject.Provider;
+import javafx.scene.Scene;
+import javafx.stage.Modality;
+import javafx.stage.Stage;
+import java.util.Map;
+import java.util.ResourceBundle;
+
+@Module
+abstract class UpdateReminderModule {
+
+	@Provides
+	@UpdateReminderWindow
+	@UpdateReminderScoped
+	static FxmlLoaderFactory provideFxmlLoaderFactory(Map<Class<? extends FxController>, Provider<FxController>> factories, DefaultSceneFactory sceneFactory, ResourceBundle resourceBundle) {
+		return new FxmlLoaderFactory(factories, sceneFactory, resourceBundle);
+	}
+
+	@Provides
+	@UpdateReminderWindow
+	@UpdateReminderScoped
+	static Stage provideStage(StageFactory factory, ResourceBundle resourceBundle) {
+		Stage stage = factory.create();
+		stage.setTitle(resourceBundle.getString("updateReminder.windowTitle"));
+		stage.setMinWidth(550);
+		stage.setMinHeight(100);
+		stage.initModality(Modality.APPLICATION_MODAL);
```

Comment:
I don't think the update window needs to have a modality (=`Modality.NONE`). If the user does not want to deal with it, it is just displayed on next start.

Reply:
I disagree. Exactly because you can simply postpone the decision by closing the window, there is no point in keeping the window open alongside other application windows.

I like to keep as many windows modal as possible, so the user isn't flooded with too many separate windows that should stick together. But maybe macOS behaves differently than Windows?